src/map.cpp

```diff
@@ -3176,54 +3185,49 @@ void map::process_items_in_vehicles( submap *const current_submap, T processor,
 }
 
 template<typename T>
-void map::process_items_in_vehicle( vehicle *cur_veh, submap *const current_submap, T processor,
-                                    std::string signal )
+void map::process_items_in_vehicle( vehicle *const cur_veh, submap *const current_submap,
+    T processor, std::string const &signal )
 {
     std::vector<int> cargo_parts = cur_veh->all_parts_with_feature(VPFLAG_CARGO, true);
     for( int part : cargo_parts ) {
         process_vehicle_items( cur_veh, part );
     }
-    auto active_items = cur_veh->active_items.get();
-    for( auto &active_item : active_items ) {
-        if( !cur_veh->active_items.has( active_item ) ) {
+
+    for( auto &active_item : cur_veh->active_items.get() ) {
+        if ( cargo_parts.empty() ) {
+            return;
+        } else if( !cur_veh->active_items.has( active_item ) ) {
             continue;
         }
 
-        // Find the cargo part and coordinates corresponding to the current active item.
-        int part_index = -1;
-        point item_location;
-        // Fetch a possibly empty item stack so we can replace it later.
-        vehicle_stack items = cur_veh->get_items( 0 );
-        for( auto part_index_candidate : cargo_parts ) {
-            vehicle_part &vp = cur_veh->parts[part_index_candidate];
-            if( active_item.location == vp.mount ) {
-                part_index = part_index_candidate;
-                item_location = cur_veh->global_pos() + vp.precalc[0];
-                items = cur_veh->get_items( part_index );
-                break;
-            }
-        }
-        if( part_index == -1 ) {
-            // Can't find a cargo part matching the active item.
-            continue;
+        auto const it = std::find_if(std::begin(cargo_parts), std::end(cargo_parts),
+            [&](int const part) { return active_item.location == cur_veh->parts[part].mount; });
+
+        if (it == std::end(cargo_parts) || *it == -1) { // TODO: Can parts even be -1??
```

Comment:
Thought so; thanks. The indices in `cargo_parts` can never be -1 then. I'll fix that up when I fix the small compile error that I missed due to compiling under a low warning level :frowning: